shotover-proxy/src/frame/cassandra.rs

```diff
@@ -476,38 +505,292 @@ impl CassandraOperation {
+        self.statements
+            .iter()
+            .map(|c| c.statement.to_string())
+            .join("; ")
+    }
+
+    /// the CassandraAST handles multiple queries in a string separated by semi-colons: `;` however
+    /// CQL only stores one query so this method only returns the first one if there are multiples.
+    pub fn parse_from_string(cql_query_str: &str) -> Self {
+        debug!("parse_from_string: {}", cql_query_str);
+        let ast = CassandraAST::new(cql_query_str);
+
+        CQL {
+            statements: ast
+                .statements
+                .iter()
+                .map(|(_, stmt)| match (ast.has_error(), stmt) {
+                    (true, CassandraStatement::Unknown(_)) => CQLStatement {
+                        statement: stmt.clone(),
+                    },
+                    (true, _) => CQLStatement {
+                        statement: CassandraStatement::Unknown(stmt.to_string()),
+                    },
+                    (false, _) => CQLStatement {
+                        statement: stmt.clone(),
+                    },
+                })
+                .collect(),
+        }
+    }
+}
+
+pub trait ToCassandraType {
+    fn from_string_value(value: &str) -> Option<CassandraType>;
+    fn as_cassandra_type(&self) -> Option<CassandraType>;
+}
+
+impl ToCassandraType for Operand {
+    fn from_string_value(value: &str) -> Option<CassandraType> {
+        // check for string types
+        if value.starts_with('\'') || value.starts_with("$$") {
+            let mut chars = value.chars();
+            chars.next();
+            chars.next_back();
+            if value.starts_with('$') {
+                chars.next();
+                chars.next_back();
+            }
+            Some(CassandraType::Varchar(chars.as_str().to_string()))
+        } else if value.starts_with("0X") || value.starts_with("0x") {
+            let mut chars = value.chars();
+            chars.next();
+            chars.next();
+            let bytes = hex::decode(chars.as_str()).unwrap();
+            Some(CassandraType::Blob(Blob::from(bytes)))
+        } else if let Ok(n) = i64::from_str(value) {
+            Some(CassandraType::Bigint(n))
+        } else if let Ok(n) = f64::from_str(value) {
+            Some(CassandraType::Double(n))
+        } else if let Ok(uuid) = Uuid::parse_str(value) {
+            Some(CassandraType::Uuid(uuid))
+        } else if let Ok(ipaddr) = IpAddr::from_str(value) {
+            Some(CassandraType::Inet(ipaddr))
+        } else {
+            None
+        }
+    }
+
+    fn as_cassandra_type(&self) -> Option<CassandraType> {
+        match self {
+            Operand::Const(value) => Operand::from_string_value(value),
+            Operand::Map(values) => Some(CassandraType::Map(
+                values
+                    .iter()
+                    .map(|(key, value)| {
+                        (
+                            Operand::from_string_value(key).unwrap(),
```

Comment:
We cant unwrap here, these are user provided strings and there is no gaurantee they are parseable.